x-pack/solutions/observability/plugins/infra/public/pages/metrics/inventory_view/components/waffle/group_name.tsx

```diff
@@ -20,41 +21,89 @@ interface Props {
   options: InfraWaffleMapOptions;
 }
 
-export class GroupName extends React.PureComponent<Props, {}> {
+export class GroupName extends React.PureComponent<Props, { a11yAnnouncement: string }> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { a11yAnnouncement: '' };
+  }
+
   public render() {
     const { group, isChild } = this.props;
-    const linkStyle = {
+    const { a11yAnnouncement } = this.state;
+    const buttonStyle = {
       fontSize: isChild ? '0.85em' : '1em',
     };
+
     return (
-      <GroupNameContainer>
-        <Inner isChild={isChild}>
-          <Name>
-            <EuiToolTip position="top" content={group.name}>
-              <EuiLink
-                style={linkStyle}
-                onClickCapture={this.handleClick}
-                data-test-subj="groupNameLink"
-              >
-                {group.name}
-              </EuiLink>
-            </EuiToolTip>
-          </Name>
-          <Count>{group.count}</Count>
-        </Inner>
-      </GroupNameContainer>
+      <>
+        <EuiScreenReaderOnly>
+          <div aria-live="polite" role="status">
+            {a11yAnnouncement}
+          </div>
+        </EuiScreenReaderOnly>
+
+        <GroupNameContainer>
+          <Inner isChild={isChild}>
+            <Name>
+              <EuiToolTip position="top" content={group.name}>
+                <EuiButtonEmpty
+                  aria-label={i18n.translate('xpack.infra.inventory.groupBySelectorButtonLabel', {
+                    defaultMessage: 'Group by {group}',
+                    values: { group: group.name },
+                  })}
+                  style={buttonStyle}
+                  onClick={this.handleClick}
+                  data-test-subj="groupNameButton"
+                >
+                  {group.name}
+                </EuiButtonEmpty>
+              </EuiToolTip>
+            </Name>
+            <Count>{group.count}</Count>
+          </Inner>
+        </GroupNameContainer>
+      </>
     );
   }
 
   private handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
     event.preventDefault();
-    const { groupBy } = this.props.options;
-    // When groupBy is empty that means there is nothing todo so let's just do nothing.
+    const {
+      options: { groupBy },
+      group,
+    } = this.props;
+
     if (groupBy.length === 0) {
+      this.setState(
```

Comment:
If we really need a role="status" element for rules, I think it should be updated once and maintain the actual status of the rules filter. Using `setTimeout` with returning value to `''` seems unnecessary to me.